src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -76,15 +77,185 @@ public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterIn
+      for (Offer offer : offers) {
+        driver.declineOffer(offer.getId());
+      }
+    } else {
+      boolean acceptedOffer = false;
+      for (Offer offer : offers) {
+        if (acceptedOffer) {
+          driver.declineOffer(offer.getId());
+        } else {
+          switch (liveState.getCurrentAcquisitionPhase()) {
+            case RECONCILING_TASKS :
+              if (liveState.reconciliationComplete()) {
+                reconcilePersistentState();
+                log.info("Current persistent state:");
+                log.info(String.format("JournalNodes: %s", persistentState.getJournalNodes()));
+                log.info(String.format("NameNodes: %s", persistentState.getNameNodes()));
+                log.info(String.format("DataNodes: %s", persistentState.getDataNodes()));
+                correctCurrentPhase();
+                driver.declineOffer(offer.getId());
+              } else {
+                log.info("Declining offers while reconciling tasks");
+                driver.declineOffer(offer.getId());
+              }
+              break;
+            case JOURNAL_NODES :
+              if (liveState.getJournalNodeSize() < conf.getJournalNodeCount()) {
+                if (tryToLaunchJournalNode(driver, offer)) {
+                  acceptedOffer = true;
+                } else {
+                  driver.declineOffer(offer.getId());
+                }
+              }
+              break;
+            case START_NAME_NODES :
+              if (tryToLaunchNameNode(driver, offer)) {
+                acceptedOffer = true;
+              } else {
+                driver.declineOffer(offer.getId());
+              }
+              break;
+            case FORMAT_NAME_NODES :
+              if (tryToLaunchNameNode(driver, offer)) {
+                acceptedOffer = true;
+              } else {
+                driver.declineOffer(offer.getId());
+              }
+              break;
+            case DATA_NODES :
+              if (tryToLaunchDataNode(driver, offer)) {
+                acceptedOffer = true;
+              } else {
+                driver.declineOffer(offer.getId());
+              }
+              break;
+          }
+        }
+      }
+    }
+  }
+
+  @Override
+  public void slaveLost(SchedulerDriver driver, SlaveID slaveId) {
+    log.info("Slave lost slaveId=" + slaveId.getValue());
   }
 
+  @Override
+  public void run() {
+    FrameworkInfo.Builder frameworkInfo = FrameworkInfo.newBuilder()
+        .setName(conf.getFrameworkName())
+        .setFailoverTimeout(conf.getFailoverTimeout())
+        .setUser(conf.getHdfsUser())
+        .setRole(conf.getHdfsRole())
+        .setCheckpoint(true);
+
+    try {
+      FrameworkID frameworkID = persistentState.getFrameworkID();
+      if (frameworkID != null) {
+        frameworkInfo.setId(frameworkID);
+      }
+    } catch (InterruptedException | ExecutionException | InvalidProtocolBufferException e) {
+      throw new RuntimeException(e);
```

Comment:
Log an error? Or do we just want to fallback to registering as a new frameworkId (probably not, if there's actually tasks they expect to recover)?